exercises/joseph_ramsay/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,182 @@
+ */
+char charToLower(const char c) {
+  if (65 <= c && c <= 90)
+    return c + 32;
+  return c;
+}
+
+/** Convert a string to lower case
+ *
+ * @param s The string to convert
+ * @return The converted string
+ */
+string lower(const string& s) {
+  string t;
+  transform(s.begin(), s.end(), back_inserter(t), ::tolower);
+  return t;
+}
+
+/** Is a character whitespace or a dash?
+ *
+ * @param c The character to test
+ * @return True if the character is whitespace or hyphen, false otherise
+ */
+bool space(const char c) { return isspace(c) || c == '-'; }
+
+/** Is not the character not whitespace or a dash?
+ *
+ * @param c The character to test
+ * @return True if the character is not whitespace or hyphen, false otherwise
+ */
+bool notSpace(const char c) { return !space(c); }
+
+/** Split a string on whitespace and hyphens
+ *
+ * @param s The string to split
+ * @return A vector of the words in the string
+ */
+std::vector<string> split(const string& s) {
+  vector<string> words;
+  auto i = s.begin();
+
+  while (i != s.end()) { // While i is not at the end of s
+    // Put i at the start of the next word, and j at the end of it
+    i = find_if(i, s.end(), notSpace);
+    string::const_iterator j = find_if(i, s.end(), space);
+
+    if (i != s.end()) // Success, so add the word to the vector
+      words.push_back(string(i, j));
+    i = j;
+  }
+  return words;
+}
+
+/** Is a character not a lowercase character?
+ *
+ * @param c The character to test
+ * @return True if the character is any other than a-z, false otherwise
+ */
+bool notLowerAlNum(const char c) {
+  return c < '0' || c > 'z' || (c > '9' && c < 'a');
+}
+
+/** Remove punctuation from a string
+ *
+ * @param s The string to remove from
+ * @return The modified string
+ */
+string noPunct(string s) {
+  string::iterator iter(remove_if(s.begin(), s.end(), notLowerAlNum));
+  s.erase(iter, s.end());
+  return s;
+}
+
+/** Get the maximum length of word in a word counter
+ * @param counter The word counter
+ * @return The length of the longest word
+ */
+int maxLen(const map<string, int> &counter) {
+  int m = 0;
+  for (const auto &element : counter){
+    m = max(m, int(element.first.length()));
+  }
+  return m;
+}
+
+/** Compare two pairs, according to their second element
+ * @param p1 The first pair
+ * @param p2 The second pair
+ * @return True if p1's second element is greater than p2's, false otherwise
+ */
+bool comparePair(const pair<string, int> &p1, const pair<string, int> &p2) {
+  return p1.second > p2.second;
+}
+
+/** Copy a map into a vector, sorted in descening order by the map's values
+ * @param counter The map to copy
+ * @return A sorted vector of pairs
+ */
+vector<pair<string, int>> sortedMap(map<string, int> &counter) {
+  vector<pair<string, int>> sorted;
+  copy(counter.cbegin(), counter.cend(), back_inserter(sorted));
+  //for_each(counter.begin(), counter.end(),
+  //         [&sorted](pair<string, int> p) { sorted.push_back(p); });
```

Comment:
Remove commented out code. If it is not used, it should not be in the source code (Saying that you will see in the coming months that there are places in Mantid where people have left commented out code, but that is mst likely by accident.).